src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,444 @@
+ *     Initialize the live restore state. Read the state from file if it exists, otherwise we start
+ *     in the log copy state and need to create the file on disk.
+ */
+int
+__wti_live_restore_init_state(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs)
+{
+    WT_DECL_RET;
+    WT_FILE_HANDLE *fh = NULL;
+
+    __wt_writelock(session, &lr_fs->state_lock);
+
+    WT_ASSERT_ALWAYS(session, lr_fs->state == WTI_LIVE_RESTORE_STATE_NONE,
+      "Attempting to initialize already initialized state!");
+
+    WT_DECL_ITEM(state_file_name);
+    WTI_LIVE_RESTORE_STATE state;
+    WT_RET(__live_restore_get_state_from_file(
+      session, lr_fs->os_file_system, lr_fs->destination.home, &state));
+
+    if (state != WTI_LIVE_RESTORE_STATE_NONE) {
+        lr_fs->state = state;
+    } else {
+        /*
+         * The state file doesn't exist which means we're starting a brand new live restore. Create
+         * the state file in the log copy state.
+         */
+        char state_to_write[128];
+        __live_restore_state_to_string(WTI_LIVE_RESTORE_STATE_LOG_COPY, state_to_write);
+
+        WT_RET(__wt_scr_alloc(session, 0, &state_file_name));
+        WT_ERR(__wt_filename_construct(session, lr_fs->destination.home,
+          WTI_LIVE_RESTORE_STATE_FILE, UINTMAX_MAX, UINT32_MAX, state_file_name));
+
+        WT_ERR(lr_fs->os_file_system->fs_open_file(lr_fs->os_file_system, (WT_SESSION *)session,
+          (char *)state_file_name->data, WT_FS_OPEN_FILE_TYPE_REGULAR,
+          WT_FS_OPEN_CREATE | WT_FS_OPEN_EXCLUSIVE, &fh));
+
+        fh->fh_write(fh, (WT_SESSION *)session, 0, 128, state_to_write);
+
+        lr_fs->state = WTI_LIVE_RESTORE_STATE_LOG_COPY;
+    }
+
+err:
+
+    __wt_writeunlock(session, &lr_fs->state_lock);
+    __wt_scr_free(session, &state_file_name);
+
+    if (fh != NULL)
+        WT_TRET(fh->close(fh, &session->iface));
+
+    return (ret);
+}
+
+/*
+ * __wti_live_restore_get_state --
+ *     Get the live restore state.
+ */
+WTI_LIVE_RESTORE_STATE
+__wti_live_restore_get_state(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_FS *lr_fs)
+{
+    WTI_LIVE_RESTORE_STATE state;
+    __wt_readlock(session, &lr_fs->state_lock);
+    state = lr_fs->state;
+    __wt_readunlock(session, &lr_fs->state_lock);
+
+    /* We initialize state on startup. This shouldn't be possible. */
+    WT_ASSERT_ALWAYS(session, state != WTI_LIVE_RESTORE_STATE_NONE, "State not initialized!");
+
+    return (state);
+}
+
+/*
+ * __wt_live_restore_delete_complete_state_file --
+ *     Provided with a folder delete any live restore state file contained within, provided that the
+ *     file is in the COMPLETE state. This function takes a *non-live restore* file system as it is
+ *     expected to be called by non-live restore file systems after live restore has completed and
+ *     WiredTiger has stated in non-live restore mode
+ */
+int
+__wt_live_restore_delete_complete_state_file(
+  WT_SESSION_IMPL *session, WT_FILE_SYSTEM *fs, const char *folder)
```

Comment:
Please `folder` -> `directory`. You use directory most often everywhere else (as well as the rest of WT)